pymongo/topology.py

```diff
@@ -692,6 +685,12 @@ def _handle_error(self, address, err_ctx):
                 # Clear the pool.
                 server.reset(service_id)
         elif issubclass(exc_type, ConnectionFailure):
+            # Ignore a handshake error if the server is behind a load balancer but
+            # the service ID is unknown. This indicates that the error happened
+            # when dialing the connection or during the MongoDB handshake, so we
+            # don't know the service ID to use for clearing the pool.
+            if self._settings.load_balanced and not service_id and not err_ctx.completed_handshake:
+                return
```

Comment:
Why limit to only connection errors?

Reply:
That was the wrong solution, I now see that we needed a way to distinguish between a handshake error and an auth error.

Reply:
Does the spec define how we’re supposed to handle an auth error?